### PR TITLE
Add documentation for exec websocket

### PR DESCRIPTION
**Pull request by backspace**
This is a continuation of #6604. You can preview it in situ [here](https://nomad-c3bdalkta.vercel.app/api-docs/allocations#exec-allocation).

### CODE REVIEW

File: website/pages/api-docs/allocations.mdx [hunk 1]
```diff
@@ -712,3 +712,83 @@ $ curl -X POST -d '{"Task": "redis" }' \
...
+
+# signaling that host closed stderr
+{"stderr": {"close": true}}
+
+# signaling process exited
+{"exited": true, "result": {"exit_code": <exit_code_int>}}
+
+# basic application-level heartbeat
+{}
+
```
Comment (DingoEatingFuzz) [inline website/pages/api-docs/allocations.mdx:793]: nit: extra newline (I tried to get rid of it with a suggestion but it didn't seem to work).

**Review by DingoEatingFuzz — APPROVED** [review]
I read through it and it makes sense to me!

The thing that wasn't clear to me was if the response frames should be appended to output or if they represent the full buffer.

I know things like `top` work but it wasn't clear what the mechanics are for line clearing. Maybe those are just xterm instructions that come through the `stdout` messages?

**Review by notnoop — APPROVED** [review]

**Comment by backspace** [comment]
> Maybe those are just xterm instructions that come through the `stdout` messages?

This is correct! Do you think this is worth mentioning?

**Comment by DingoEatingFuzz** [comment]
I don't think it would hurt to add it.

**Comment by backspace** [comment]
> I don't think it would hurt to add it.

👍 I added an [example request and response](https://nomad-3kdta0oca.vercel.app/api-docs/allocations#sample-request-and-response) to provide some explanation of the content of frames, thoughts?

**Review by DingoEatingFuzz — APPROVED** [review]
I like it! That sample gives just enough guidance on what to expect while also introducing some vocab someone can do further research on.

:shipit:

**Pull request merged**